lib/dsc-lib/src/functions/mod.rs

```diff
@@ -289,18 +297,79 @@ impl FunctionDispatcher {
         function.invoke(args, context)
     }
 
+    /// Special handler for lambda() function calls.
+    /// Creates a Lambda object and stores it in Context with a unique ID.
+    ///
+    /// # Arguments
+    ///
+    /// * `args` - Raw FunctionArg list (unevaluated)
+    /// * `context` - Context to store the lambda in
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if the lambda syntax is invalid.
+    pub fn invoke_lambda(&self, args: &Option<Vec<crate::parser::functions::FunctionArg>>, context: &Context) -> Result<Value, DscError> {
+        use crate::parser::functions::{FunctionArg, Lambda};
+        use uuid::Uuid;
+        
+        let Some(args) = args else {
+            return Err(DscError::Parser(t!("functions.lambda.requiresArgs").to_string()));
+        };
+
+        if args.len() < 2 {
+            return Err(DscError::Parser(t!("functions.lambda.requiresParamAndBody").to_string()));
+        }
+
+        // All arguments except the last must be string values (parameter names)
+        let mut parameters = Vec::new();
+        for arg in args.iter().take(args.len() - 1) {
+            match arg {
+                FunctionArg::Value(Value::String(s)) => {
+                    parameters.push(s.clone());
+                },
+                _ => {
+                    return Err(DscError::Parser(t!("functions.lambda.paramsMustBeStrings").to_string()));
+                }
+            }
+        }
+
+        // Last argument is the body expression
+        let body_expr = match &args[args.len() - 1] {
+            FunctionArg::Expression(expr) => expr.clone(),
+            _ => {
+                return Err(DscError::Parser(t!("functions.lambda.bodyMustBeExpression").to_string()));
+            }
+        };
+
+        // Create Lambda and store in Context with unique ID
+        let lambda = Lambda {
+            parameters,
+            body: body_expr,
+        };
+        
+        let lambda_id = format!("__lambda_{}", Uuid::new_v4());
+        context.lambdas.borrow_mut().insert(lambda_id.clone(), lambda);
+
+        // Return the ID as a string value
+        Ok(Value::String(lambda_id))
+    }
```

Comment:
Instead of special handling it this way, you should have a new ProcessMode::Lambda and just have the `lambda()` function do this only when process mode is correct other return an error.  This keeps it consistent with modes.